src/client/java/teammates/client/scripts/statistics/FileStore.java

```diff
@@ -0,0 +1,212 @@
+package teammates.client.scripts.statistics;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.spec.SecretKeySpec;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import teammates.common.util.Config;
+import teammates.common.util.StringHelper;
+import teammates.test.driver.FileHelper;
+
+/**
+ * File storage service that saves stats/cache to encrypted/unencrypted files.
+ */
+public class FileStore {
```

Comment:
The way this class is constructed, it can be a general purpose file storage and not just for this statistics purpose, don't you think?

Reply:
Is it too early to make the generalisation? This is the first place where file is stored in encrypted format.